src/gfx_layout_icu.cpp

```diff
@@ -381,6 +381,32 @@ std::vector<ICURun> ItemizeStyle(std::vector<ICURun> &runs_current, FontMap &fon
 	return new ICUParagraphLayout(runs, buff, length);
 }
 
+struct BreakIteratorInstance
+{
+	std::unique_ptr<icu::BreakIterator> break_iterator;
+	std::string isocode;
+
+	static BreakIteratorInstance instance;
```

Comment:
Won't making this instance thread local, make it even thread safer?
```suggestion
	thread_local static BreakIteratorInstance instance;
```
Not sure whether this yields the result you're looking for though. Also not sure what the performance impact of `thread_local` is. It's just something to consider.

Reply:
I did actually try this but it made it really hard to debug it because the debugger is never in the right thread, or something :D